core/imt/inc/ROOT/TExecutor.hxx

```diff
@@ -0,0 +1,494 @@
+// @(#)root/thread:$Id$
+// Author: Xavier Valls September 2020
+
+/*************************************************************************
+ * Copyright (C) 1995-2006, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+#ifndef ROOT_TExecutor
+#define ROOT_TExecutor
+
+#include <ROOT/RMakeUnique.hxx>
+#include "ROOT/TExecutorCRTP.hxx"
+#include "ROOT/TSequentialExecutor.hxx"
+#ifdef R__USE_IMT
```

Comment:
Where is `R__USE_IMT` defined, do you need to `#include "ROOT/RConfig.hxx"` or something?